include/nn/socket.h

```diff
@@ -25,6 +26,8 @@ using InAddr = ::in_addr;
 Result Initialize(void* pool, ulong poolSize, ulong allocPoolSize, int concurLimit);
 Result Finalize();
 s32 SetSockOpt(s32 socket, s32 socketLevel, s32 option, const void*, u32 len);
+s32 Poll(pollfd *fds, unsigned long nfds, s32 timeout);
```

Comment:
Can the unsigned long just be nfds_t as to match how poll is done in UNIX
https://man7.org/linux/man-pages/man2/poll.2.html

Reply:
@ThePixelGamer no, the mangled name won't link if you use nfds_t, i tried

Reply:
mangled name? the type should exist seeing how you're including poll.h, am I missing something?

Reply:
Also, nfds_t is a typedef so it won't even show up in the mangled symbol.

Reply:
@leoetlino it's typedefed as an unsigned int and not an unsigned long, so the mangled name changes

Reply:
musl (which is the libc NintendoSDK uses) typedefs it to an unsigned long.

Reply:
@leoetlino i tried, and it seemed to mangle it as `j` and not `m`. not sure why

Reply:
```
		 ::= j	# unsigned int
		 ::= l	# long
		 ::= m	# unsigned long
```

Sounds like your header include path is set up incorrectly and you might be using your host system's poll.h (which typedefs it to unsigned int?) rather than the one from musl.